hadoop-ozone/s3gateway/src/main/java/org/apache/hadoop/ozone/s3/endpoint/ObjectEndpoint.java

```diff
@@ -233,11 +255,19 @@ public Response put(
           .equals(headers.getHeaderString("x-amz-content-sha256"))) {
         body = new SignedChunksInputStream(body);
       }
+      long putLength = 0;
+      if (datastreamEnabled && !enableEC) {
```

Comment:
How about we use Streaming only if it has more than one chunk?
```java
      if (datastreamEnabled && !enableEC && length > chunkSize) {
```

Reply:
sure.